src/angularjs/src/app/components/scoremetadata.service.js

```diff
@@ -8,6 +8,47 @@
 (function() {
     'use strict';
 
+    /**
+     Sort metrics in metadata by category, with total first in the category.
+     Add subscoreClass property to metrics that should be indented under the total.
+     */
+    function orderCategories(response) {
```

Comment:
Ah, I didn't know there was explicit ordering applied to the metadata response from the API.

Relying on the ordering of the response to be stable will also mean assuming that the ordering matches the structure we need (grouped by category, with summary metric first). This task has those implicit ordering assumptions.

Dropping the category for summary scores would potentially break any other use of categories.

If we can rely on a stable sort order from the API that matches those assumptions, then this client-side sub-category identification logic can be simplified somewhat to just iterate through once.